### PR TITLE
doc(api): define contract of resource listing endpoint

### PR DESCRIPTION
**Pull request by kduret**
## Description

Define contract of resource listing endpoint to allow backend and frontend to work efficiently

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [x] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

### CODE REVIEW

File: doc/API/centreon-api-v2.yaml [hunk 1]
```diff
@@ -1415,6 +1450,137 @@ components:
...
+              description: |
+                resource status code
+                * `0` - UP or OK
+                * `1` - DOWN or WARNING
+                * `2` - UNREACHABLE or CRITICAL
+                * `3` - UNKNOWN
+                * `4` - PENDING
+              example: 0
+            name:
+              type: integer
```
Comment (bdauria) [inline doc/API/centreon-api-v2.yaml]: should be string

File: doc/API/centreon-api-v2.yaml [hunk 2]
```diff
@@ -1415,6 +1450,137 @@ components:
...
+          type: string
+          description: "Resource name"
+          example: "Ping"
+        details_url:
+          type: string
+          description: "Resource url to details page"
+          example: "/monitoring/host/12/service/12"
+        icon:
+          type: object
+          properties:
```
Comment (vhr) [inline doc/API/centreon-api-v2.yaml]: there is already has an object for that and the structure is a little bit different https://github.com/centreon/centreon/blob/master/src/Centreon/Domain/Entity/Image.php#L35

```suggestion
          properties:
            id:
              type: integer
              format: int64
              description: "ID of the image"
              example: 12
```

File: doc/API/centreon-api-v2.yaml [hunk 3]
```diff
@@ -1415,6 +1450,137 @@ components:
...
+            name:
+              type: string
+              description: "Name of the icon"
+              example: "memory"
+            url:
```
Comment (vhr) [inline doc/API/centreon-api-v2.yaml]: url -> preview https://github.com/centreon/centreon/blob/master/src/Centreon/Domain/Entity/Image.php#L108
```suggestion
            preview:
```

File: doc/API/centreon-api-v2.yaml [hunk 4]
```diff
@@ -1415,6 +1450,137 @@ components:
...
+                  description: "Name of the icon"
+                  example: "linux"
+                url:
+                  type: string
+                  description: "Url of the icon"
+                  example: "/media/linux.png"
+        status:
+          type: object
+          properties:
+            code:
```
Comment (vhr) [inline doc/API/centreon-api-v2.yaml]: the object identifier has to be `id`

```suggestion
            id:
```

File: doc/API/centreon-api-v2.yaml [hunk 5]
```diff
@@ -1415,6 +1450,137 @@ components:
...
+          type: boolean
+          description: "Indicates whether resource is in downtime"
+          example: false
+        acknowledged:
+          type: boolean
+          description: "Indicates whether resource is acknowledged"
+          example: false
+        severity:
+          type: object
+          properties:
```
Comment (vhr) [inline doc/API/centreon-api-v2.yaml]: the object must have an identifier

```suggestion
          properties:
            id:
              type: integer
              format: int64
              description: "ID of the severity"
              example: 12
```

File: doc/API/centreon-api-v2.yaml [hunk 6]
```diff
@@ -1415,6 +1450,137 @@ components:
...
+              type: string
+              description: "Name of the icon"
+              example: "memory"
+            url:
+              type: string
+              description: "Url of the icon"
+              example: "/media/memory.png"
+        parent:
+          type: object
+          properties:
```
Comment (vhr) [inline doc/API/centreon-api-v2.yaml]: Do we need the type property in the parent object?

File: doc/API/centreon-api-v2.yaml [hunk 7]
```diff
@@ -1415,6 +1450,137 @@ components:
...
+              example: "UP"
+        in_downtime:
+          type: boolean
+          description: "Indicates whether resource is in downtime"
+          example: false
+        acknowledged:
+          type: boolean
+          description: "Indicates whether resource is acknowledged"
+          example: false
+        severity:
```
Comment (vhr) [inline doc/API/centreon-api-v2.yaml]: How about to make the severity object with the same structure as the icon object?

File: doc/API/centreon-api-v2.yaml (hunk 1 above)
Comment (kduret) [inline doc/API/centreon-api-v2.yaml]:
```suggestion
              type: string
```

File: doc/API/centreon-api-v2.yaml [hunk 8]
```diff
@@ -1415,6 +1450,142 @@ components:
...
+    Monitoring.ResourceMain:
+      type: object
+      properties:
+        type:
+          type: string
+          description: "Type of the resource"
+          example: "service"
+        id:
+          type: integer
+          format: int64
```
Comment (kduret) [inline doc/API/centreon-api-v2.yaml]:
```suggestion
          format: int32
```

File: doc/API/centreon-api-v2.yaml [hunk 9]
```diff
@@ -1415,6 +1450,147 @@ components:
...
+        acknowledged:
+          type: boolean
+          description: "Indicates whether resource is acknowledged"
+          example: false
+        severity:
+          type: object
+          properties:
+            id:
+              type: integer
+              format: int64
```
Comment (kduret) [inline doc/API/centreon-api-v2.yaml]:
```suggestion
              format: int32
```

File: doc/API/centreon-api-v2.yaml [hunk 10]
```diff
@@ -1415,6 +1450,147 @@ components:
...
+        details_url:
+          type: string
+          description: "Resource url to details page"
+          example: "/monitoring/host/12/service/12"
+        icon:
+          type: object
+          properties:
+            id:
+              type: integer
+              format: int64
```
Comment (kduret) [inline doc/API/centreon-api-v2.yaml]:
```suggestion
              format: int32
```

File: doc/API/centreon-api-v2.yaml (hunk 7 above)
Comment (kduret) [reply doc/API/centreon-api-v2.yaml]: we use "id", "name" and "url" for those two objects

File: doc/API/centreon-api-v2.yaml (hunk 6 above)
Comment (kduret) [reply doc/API/centreon-api-v2.yaml]: don't know but it seems to be done like that for all objects

File: doc/API/centreon-api-v2.yaml (hunk 4 above)
Comment (kduret) [reply doc/API/centreon-api-v2.yaml]: it is not really an identifier, cause 0 is used for "UP" and "OK"

**Pull request merged**